app/server/runtime/src/main/java/io/syndesis/server/runtime/GraphQLConfiguration.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.runtime;
+
+import static graphql.schema.idl.RuntimeWiring.newRuntimeWiring;
+import static io.syndesis.common.util.IOStreams.readText;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import io.syndesis.common.model.integration.Step;
+import io.syndesis.server.endpoint.v1.handler.integration.IntegrationHandler;
+
+/**
+ * Used to setup a GraphQL endpoint.
+ */
+@Configuration
+//@ConditionalOnProperty("features.api.graphql.enabled")
+public class GraphQLConfiguration {
+
+    @Autowired
+    IntegrationHandler integrationHandler;
+
+    @Bean
+    GraphQLSchema schema() throws IOException {
+
+        String schema;
+        try (InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("schema.graphqls") )  {
+            schema = readText(is);
+        }
+
+        SchemaParser schemaParser = new SchemaParser();
+        TypeDefinitionRegistry typeDefinitionRegistry = schemaParser.parse(schema);
+
+        SchemaGenerator schemaGenerator = new SchemaGenerator();
+        return schemaGenerator.makeExecutableSchema(typeDefinitionRegistry, newRuntimeWiring()
+
+            // Entities who's json shape matches that GraphQL schema,
+            // just need a a bit of code like this to load the data for the query.
+            .type("Query", builder -> builder
+                .dataFetcher("integrations", env->{
+                    return integrationHandler.list(null);
```

Comment:
I'd avoid this coupling of handler to a handler, we have already done this and it has caused only issues.

Reply:
can you point me at what those issues were?

Reply:
#5995 and #6111